model/jpa/src/main/java/org/keycloak/models/workflow/JpaWorkflowStateProvider.java

```diff
@@ -57,17 +59,24 @@ public ScheduledStep getScheduledStep(String workflowId, String resourceId) {
     public void scheduleStep(Workflow workflow, WorkflowStep step, String resourceId) {
         WorkflowStateEntity.PrimaryKey pk = new WorkflowStateEntity.PrimaryKey(resourceId, workflow.getId());
         WorkflowStateEntity entity = em.find(WorkflowStateEntity.class, pk);
+
+        Long workflowStepTimeOffset = session.getAttribute(WORKFLOW_FIRST_STEP_TIME_OFFSET, Long.class);
+        long scheduledStepTimestamp = (workflowStepTimeOffset == null) ?
+                Time.currentTimeMillis() + step.getAfter() :
+                Time.currentTimeMillis() + step.getAfter() + workflowStepTimeOffset;
+        session.removeAttribute(WORKFLOW_FIRST_STEP_TIME_OFFSET);
```

Comment:
Wouldn't it be better to have a `scheduleStep` variant that has the offset as a parameter? Using a session attribute is a bit odd, isn't it?